### DISCUSSION TITLE
Add more subscriptions event for easy customize

### DISCUSSION THREAD

**Discussion by thanhlmm** [tradingview/lightweight-charts#846] [Ideas]
After few months working with `lightweight-charts` and many others charting lib, I need to face many custom types, and there are lots of issues regarding to it like #278, #408, #694,...etc

The problem comes from when we need to customize:
- The charting is supported that customizes but the aspect is fixed as the lib (Eg: price label shape, tooltip shape, ...)
- Or it might not support it

So we end up trying another charting library or finding a hacky way to do it

Which `lightweight-charts`, I figure out that we can do some customize by subscribing to its event and draw a layer on top of it, by doing so, we have lots of customizing space, ideas to build on top of it (Eg: Customize crossmark tooltip, mark with html supported, minimap,...)

So I think we should add more subscriptions events so that the customization is more open. Few ideas on my mind:
- Panel size change
- Price scale change
- Hover event
- 
By doing so, I think we can solve lots of issues by giving dev opportunities to hack around it. It likes photoshop with a customize layer on top of it

Of course if might come with some tradeoff, let's discuss it here

**Comment by timocov** [comment]
> Panel size change

Do you mean the whole chart or a particular pane?

> Price scale change

Yeah agree, since the size adjustments is done inside the library it would be good to have it. We have [a getter for width](https://github.com/tradingview/lightweight-charts/blob/master/src/api/iprice-scale-api.ts#L24) of price scale already btw.

> Hover event

Don't we have something like that already? I thought you could subscribe on hover event externally (by adding an even listener to container), couldn't you?

**Comment by thanhlmm** [comment]
> Do you mean the whole chart or a particular pane?
I mean particular panel like Axis panel, Time axis panel

> Yeah agree, since the size adjustments is done inside the library it would be good to have it. We have a getter for width of price scale already btw.

Awesome, I have to use dom to check the width 😅

> Don't we have something like that already? I thought you could subscribe on hover event externally (by adding an event listener to container), couldn't you?

We're having crosshead `CrosshairMove` works like hover event on the chart so I think it would be nice if we have similar event when we move the mouse on the axis. Eg case: Hovering on the price axis to set price alert,...

**Reply by timocov** [reply]
> I think it would be nice if we have similar event when we move the mouse on the axis. Eg case: Hovering on the price axis to set price alert,...

So it seems that you want to get some specific events rather than just "hover event"? I think you can create a feature request with details about your request.

**Reply by thanhlmm** [reply]
To be honest, this is just an idea, I don't have this case in real life. However, I think maybe, if someone has the issue related to that, we can start to do
